routes/index.js

```diff
@@ -405,7 +405,7 @@ router.get('/find', (req, res, next) => {
             opts.previous = cursorValue;
         }
 
-        MongoPaging.find(db.client.collection('mids'), opts, (err, result) => {
+        MongoPaging.find(db.client.collection('mids'), opts).then((result) => {
```

Comment:
Could you explain this change? You are replacing the callback function with a Promise, but why? Also, the error reponse gets unhandled in this way.

Reply:
I've been searching on mongo-cursor-pagination and cannot find **any** thing about find having being changed (  seems to have not been updated in 2 years ) ... 
And i can see all dependencies in  project.json in this project have a fixed version's including mongodb, so I don't know what part has changed you wouldn't have caught ...
I guess there must be something wrong on my system then?
But if i look at find.js in mongo-cursor-pagination it's declared as 
```
module.exports = async function(collection, params) {
```
If i take a clean version of this project and do a search it hangs for ever after `return response`; inside MongoPaging.find and never returns. If i change it to use .then as if MongoPaging.find returns a promise, it completes and searching is working again.
hence why I assumed the code was wrong.
But you are absolutely correct, a massive error i forgot to add .catch to handle calling next on error.
I have added that now

Reply:
Ok. Remove the `if (err) {` block as well from the `then()` function as there is no `err` variable anymore when using promises, then it should be fine.

Reply:
done